src/ChainState.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from "react";
+
+import { Grid, Form, Dropdown, Button, Input } from "semantic-ui-react";
+
+export default function Metadata(props) {
+  const { api } = props;
+
+  const [modulesList, setModulesList] = useState([]);
+  const [output, setOutput] = useState("");
+  const [storageItemsList, setStorageItemsList] = useState([]);
+
+  const initialState = {
+    module: "",
+    storageItem: "",
+    input: ""
+  };
+  const [formState, setFormState] = useState(initialState);
+  const { module, storageItem, input } = formState;
+
+  useEffect(() => {
+    let queries = api.query;
+    let modules = Object.keys(queries).map(module => ({
+      key: module,
+      value: module,
+      text: module
+    }));
+
+    setModulesList(modules);
+  }, [api]);
+
+  useEffect(() => {
+    if (module !== "") {
+      let queries = eval("api.query." + module + ".valueOf()");
+      let storageItems = Object.keys(queries).map(storage => ({
+        key: storage,
+        value: storage,
+        text: storage
+      }));
+
+      setStorageItemsList(storageItems);
+    }
+  }, [api, module]);
+
+  const onChange = (_, data) => {
+    setFormState(formState => {
+      return {
+        ...formState,
+        [data.state]: data.value
+      };
+    });
+  };
+
+  const runQuery = async () => {
+    let result = await eval(
+      "api.query." + module + "." + storageItem + "('" + input + "')"
```

Comment:
Can just be `await api.query[module][storageItem](input)` - no eval

Reply:
Is there an easy way to learn if the query needs an input?

i.e. free balance expects a user account, while sudo key needs no input